File: api/ejecutar_octave.py
import json
import subprocess

def handler(event, context):
    data = json.loads(event['body'])

    monto = data['monto']
    plazo = data['plazo']
    tasas = data['tasas']

    result = subprocess.run(
        ['octave', '--eval',
         f"addpath('../backend_octave'); calcular_cuota({monto}, {plazo}, [{tasas}]);"],
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE
    )

    # Aquí deberías leer el archivo JSON de salida o los datos generados por Octave
    output = {
        "cuota": 1234.56,
        "total": 43210.00,
        "grafico": ""  # base64 del gráfico generado
    }

    return {
        "statusCode": 200,
        "body": json.dumps(output),
        "headers": {
            "Content-Type": "application/json"
        }
    }

File: netlify/functions/ejecutar_octave.js
exports.handler = async function (event, context) {
  const body = JSON.parse(event.body);

  return {
    statusCode: 200,
    body: JSON.stringify({
      cuota: 1234.56,
      total: 43210.00,
      grafico: ""  // Podrías usar base64 de un gráfico real más adelante
    }),
    headers: {
      "Content-Type": "application/json"
    }
  };
};
